### PR TITLE
Add ST_LineMerge

### PR DESCRIPTION
**Pull request by imdany**
## Is this PR related to a proposed Issue?

[Implement complete SQL/MM and PostGIS functions in GeoSparkSQL](https://github.com/DataSystemsLab/GeoSpark/issues/200)

## What changes were proposed in this PR?

Added ST_LineMerge, a function that works on MultiLineStrings joining the component forming a single LineString if they match the appropriate condition.

## How was this patch tested?

Unit Testing and Jar tested in Databricks environment

## Did this PR include necessary documentation updates?
Yes.

### CODE REVIEW

**Comment by jiayuasu** [comment]
@imdany Can you please fix the conflicting files?

I just merged a couple of pending PRs. So this PR seems to be conflicting with others.

Thank you very much!

File: docs/api/sql/GeoSparkSQL-Function.md [hunk 1]
```diff
@@ -269,6 +269,18 @@ SELECT ST_GeometryType(polygondf.countyshape)
 FROM polygondf
 ````
 
+## ST_LineMerge
+
+Introduction: Returns a LineString formed by sewing together the constituent line work of a MULTILINESTRING.
+
+Note: Only works for MULTILINESTRING. Using other geometry will return a GEOMETRYCOLLECTION EMPTY. If the MultiLineString can't be merged, the original MULTILINESTRING is returned.
+
```
Comment (jiayuasu) [inline docs/api/sql/GeoSparkSQL-Function.md:277]: Please add - Since: `v1.3.2`

Comment (jiayuasu) [inline docs/api/sql/GeoSparkSQL-Function.md:277]: Please add - Since: v1.3.2

Comment (imdany) [reply docs/api/sql/GeoSparkSQL-Function.md:277]: Ups... this must have gone missing when I was fixing the conflicts. It's fixed now.

**Comment by jiayuasu** [comment]
@imdany Thanks for your contribution!

**Pull request merged**